### ISSUE TITLE
[resign] allow provisioning identifier as parameter instead of path to file

### ISSUE THREAD

**Issue by zxcq99** [fastlane/fastlane#9338]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [x ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
 It would be nice to be able to provide identifier of provisioning profile instead of a path to .mobileprovision file and fastlane/resign will automatically find the matching provisioning profile installed on machine.

**Comment by PetrosIchor** [comment]
From my understanding of the resign action , that would make it overly complicated, brittle, and break a ton of other users setups.  The action would still need to know where to look for the profile, then crack each one open and identity which one it needed.  

For any user who uses custom paths, that's an extra parameter they would be required to account for.

Resign is heavily tied to sigh/match.  Those tools simplify building new actions and plugins.  They are a hub for iOS signing functionality that makes the fastlane so powerful. 

In house, I use match. Recently I had a high need to use the resign action for the first time. A series of certificates/profiles were accidentally revoked and I needed to redeploy multiple apps with new signing (the signing also needed to be generated). 

I don't think it took me more than 20 minutes to create a script, drop in into my Jenkins pipeline, and redeploy.  Adopting fastlanes signing paradigm, I was able to shave days worth of critical infrastructure work down to 20 minutes.  All that was left was to leave a tab open with build progress bars and to keep an eye on slack for a failure notifications. 

In my eyes, resign is a last resort tool.  When you package an application, you shouldn't expect to open it again. Every time you touch a completed product, you increase the risk of issues. Resign is setup to be specific.  You have a need to break open something sealed, resign wants to make sure it uses the exact assets you require. Otherwise, it might break something. 

Just my 2cents.  Anyone elses thoughts?  I might be off on my understanding of resign.
